File: app/backend/models/usuario_model.py
from sqlalchemy import Column, Integer, String, Boolean, Date, DateTime
from backend.database.connection import Base

class Usuario(Base):
    __tablename__ = "Usuario"

    IdUsuario = Column(Integer, primary_key=True, index=True, autoincrement=True)
    NombreCompleto = Column(String(60), nullable=False)
    NombreUsuario = Column(String(30), unique=True, nullable=False)
    Email = Column(String(100), unique=True, nullable=False)
    Password = Column(String(255), nullable=False)  # Aumentado para almacenar hash
    Rol = Column(String(50), nullable=False)
    Activo = Column(Boolean, default=False, nullable=False)
    FechaCreacion = Column(Date, nullable=False)
    UltimaConexion = Column(DateTime, nullable=True)
    Telefono = Column(String(25), nullable=True)
    Observacion = Column(String(200), nullable=True)
    Descripcion = Column(String(200), nullable=True)

File: app/frontend/src/app/features/usuarios/models/usuario.model.ts
export interface Usuario {
  IdUsuario: number;
  NombreCompleto: string;
  NombreUsuario: string;
  Email: string;
  Rol: string;
  Activo: boolean;
  FechaCreacion: string;
  UltimaConexion?: string;
  Telefono?: string;
  Observacion?: string;
  Descripcion?: string;
}

export interface UsuarioCreate {
  NombreCompleto: string;
  NombreUsuario: string;
  Email: string;
  Password: string;
  Rol: string;
  Activo: boolean;
  FechaCreacion?: string;  // Opcional porque se genera automáticamente
  UltimaConexion?: string;
  Telefono?: string;
  Observacion?: string;
  Descripcion?: string;
}

export interface UsuarioUpdate {
  NombreCompleto?: string;
  NombreUsuario?: string;
  Email?: string;
  Password?: string;
  Rol?: string;
  Activo?: boolean;
  FechaCreacion?: string;
  UltimaConexion?: string;
  Telefono?: string;
  Observacion?: string;
  Descripcion?: string;
}

export interface UsuarioLogin {
  NombreUsuario: string;
  Password: string;
}

export interface CambiarPasswordRequest {
  password_actual: string;
  password_nueva: string;
}

export interface ActivarUsuarioRequest {
  activo: boolean;
}